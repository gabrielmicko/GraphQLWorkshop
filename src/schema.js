/**
 * TASK 2
 *
 * Import makeExecutableSchema, resolvers, typeDefs
 * Call & export makeExecutableSchema, pass the imported variables
 */
