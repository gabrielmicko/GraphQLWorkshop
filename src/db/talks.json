[
  {
    "id": "1",
    "date": "Aug 3 - 10:30",
    "duration": 50,
    "talkType": "keynote",
    "talkTitle": "JavaScript: clouds on the horizon",
    "speakerId": "1"
  },
  {
    "id": "2",
    "date": "Aug 3 - 11:20",
    "duration": 40,
    "talkType": "talk",
    "talkTitle": "Custom RxJS operators",
    "speakerId": "2"
  },
  {
    "id": "3",
    "date": "Aug 3 - 12:30",
    "duration": 40,
    "talkType": "talk",
    "talkTitle": "From code to pixels in a Blink",
    "speakerId": "3"
  },
  {
    "id": "4",
    "date": "Aug 3 - 13:10",
    "duration": 50,
    "talkType": "talk",
    "talkTitle": "Vue + GraphQL",
    "speakerId": "4"
  },
  {
    "id": "5",
    "date": "Aug 3 - 15:20",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "React with Blackjack and Hooks",
    "speakerId": "5"
  },
  {
    "id": "6",
    "date": "Aug 3 - 15:30",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Accessibility through the eyes of users",
    "speakerId": "6"
  },
  {
    "id": "7",
    "date": "Aug 3 - 15:40",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Tensorflow in Action",
    "speakerId": "7"
  },
  {
    "id": "8",
    "date": "Aug 3 - 15:50",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Language Independent Validation Rules 2.0",
    "speakerId": "8"
  },
  {
    "id": "9",
    "date": "Aug 3 - 16:00",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Reactive programming. Svelte way",
    "speakerId": "9"
  },
  {
    "id": "10",
    "date": "Aug 3 - 16:10",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "ML + AR in js (to be or not to be)",
    "speakerId": "10"
  },
  {
    "id": "11",
    "date": "Aug 3 - 16:50",
    "duration": 40,
    "talkType": "talk",
    "talkTitle": "Fast future with WebAssembly",
    "speakerId": "11"
  },
  {
    "id": "12",
    "date": "Aug 3 - 17:30",
    "duration": 40,
    "talkType": "keynote",
    "talkTitle": "A universal fusion of the team",
    "speakerId": "12"
  },
  {
    "id": "13",
    "date": "Aug 4 - 11:00",
    "duration": 150,
    "talkType": "workshop",
    "talkTitle": "Building GraphQL Server with JS",
    "speakerId": "13"
  },
  {
    "id": "14",
    "date": "Aug 4 - 14: 30",
    "duration": 90,
    "talkType": "workshop",
    "talkTitle": "React with Blackjack and Hooks",
    "speakerId": "5"
  },
  {
    "id": "15",
    "date": "Aug 4 - 17:00",
    "duration": 90,
    "talkType": "workshop",
    "talkTitle": "Reactive programming. Svelte way",
    "speakerId": "9"
  }
]
