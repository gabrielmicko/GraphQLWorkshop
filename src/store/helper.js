import fs from 'fs';
import low from 'lowdb';
import FileSync from 'lowdb/adapters/FileSync';

const database = low(new FileSync(__dirname + '/../../temp/database.json'));

/**
 * Returns a proper table name based on environment.
 * @param  {String} tableName
 * @return {String} tableName
 */
const getTable = tableName => {
  if (process.env.NODE_ENV === 'test') {
    return 'test_' + tableName;
  }
  return tableName;
};

/**
 * Sets the default speakers and talks in the test database.
 * @param  {Array} speakers
 * @param  {Array} talks
 */
const setTestDefaults = (speakers, talks) => {
  database.set('test_speakers', speakers).write();
  database.set('test_talks', talks).write();
};

/**
 * Sets the default speakers and talks in the test and live database.
 * @param  {Array} speakers
 * @param  {Array} talks
 */
const setDefaults = (speakers, talks) => {
  database.set('speakers', speakers).write();
  database.set('talks', talks).write();
  setTestDefaults(speakers, talks);
};

/**
 * TASK 5
 * Checks if data is data is Array or Object.
 * If Array iterates through it and pushes it to the database..
 * If Object it pushes it to the database.
 * Returns with the inserted data, Array of Objects or Object.
 * @param  {String} tableName Table name, where you insert/update/remove/gets data.
 * @param  {Array|Object} data Array containing Objects or a single Object.
 * @return {Array|Object} Retunrs the inserted data.
 */
const insertData = (tableName, data) => {
  let table = getTable(tableName);
};

/**
 * TASK 5
 * Get all data from a table according to a filter object.
 * @param  {String} tableName Table name, where you insert/update/remove/get data.
 * @param  {Object} filters Object  containing the filter (ie. {speakerId: 10}, check: filter()).
 * @return {Array} Returns the filtered data set.
 */
const getByFilters = (tableName, filters = {}) => {
  let table = getTable(tableName);
};

/**
 * TASK 5
 * Get all the data from a table.
 * @param  {String} tableName Table name, where you insert/update/remove/get data.
 * @return {Array} Returns all the data in the table.
 */
const getAll = tableName => {
  let table = getTable(tableName);
};

/**
 * TASK 5
 * Removes everything from the table.
 * @param  {String} tableName Table name, where you insert/update/remove/get data.
 */
const wipeTable = tableName => {
  let table = getTable(tableName);
};

/**
 * TASK 5
 * Accepts list of tables (strings), calls wipeTable on each iteration.
 * @param  {Array} tables Array containing tables (strings), which you want to wipe (ie: ["speakers", "talks"]).
 */
const wipeTables = tables => {};

export { getTable, insertData, wipeTables, getAll, getByFilters, wipeTable, setDefaults, setTestDefaults };
