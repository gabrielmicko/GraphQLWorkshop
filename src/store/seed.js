/*
⚠️ Do not edit this file
*/

import { setDefaults } from './helper';
console.log('✏️ - Database created.');
import speakers from '../db/speakers.json';
import talks from '../db/talks.json';

async function seedDatabase() {
  console.log('💣 - Setting defaults.');
  setDefaults(speakers, talks);
  console.log('💾 - Speakers and talks have been uploaded.');
  console.log('🎉 - Work done, enjoy 🍦');
}

seedDatabase();
