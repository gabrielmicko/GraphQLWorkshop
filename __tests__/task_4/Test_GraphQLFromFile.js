import { graphql } from 'graphql';
import schema from '../../src/schema';
import { setTestDefaults } from '../../src/store/helper';
import speakers from '../../src/db/speakers.json';
import talks from '../../src/db/talks.json';

describe('tests task 4', () => {
  beforeEach(() => {
    setTestDefaults(speakers, talks);
  });

  test('GraphQL server should return one speaker by ID.', () => {
    expect.assertions(1);
    const query = /* GraphQL */ `
      {
        speakers(id: "13") {
          id
          name
          twitter
          github
          linkedin
          bio
          talks {
            id
            date
            duration
            talkType
            talkTitle
            speakerId
          }
        }
      }
    `;
    return graphql(schema, query).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('GraphQL server should return one speaker by name.', () => {
    expect.assertions(1);
    const query = /* GraphQL */ `
      {
        speakers(name: "Gabriel Mičko") {
          id
          name
          twitter
          github
          linkedin
          bio
          talks {
            id
            date
            duration
            talkType
            talkTitle
            speakerId
          }
        }
      }
    `;
    return graphql(schema, query).then(result => {
      expect(result).toMatchSnapshot();
    });
  });
});
