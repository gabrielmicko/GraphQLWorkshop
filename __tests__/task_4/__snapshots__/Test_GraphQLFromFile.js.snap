// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testsuu task 4 GraphQL server should return one speaker by ID. 1`] = `
Object {
  "data": Object {
    "speakers": Array [
      Object {
        "bio": "I am enthusiastic front-end developer who loves the web and new technologies around it. I am tech speaker, contributor, volunteer at Mozilla. I believe in the mission to ensure the Internet is a global public resource, open and accessible to all.",
        "github": "https://github.com/gabrielmicko",
        "id": "13",
        "linkedin": null,
        "name": "Gabriel Mičko",
        "talks": Array [
          Object {
            "date": "Aug 4 - 11:00",
            "duration": 150,
            "id": "13",
            "speakerId": "13",
            "talkTitle": "Building GraphQL Server with JS",
            "talkType": "workshop",
          },
        ],
        "twitter": "https://twitter.com/gabriel_micko",
      },
    ],
  },
}
`;

exports[`testsuu task 4 GraphQL server should return one speaker by name. 1`] = `
Object {
  "data": Object {
    "speakers": Array [
      Object {
        "bio": "I am enthusiastic front-end developer who loves the web and new technologies around it. I am tech speaker, contributor, volunteer at Mozilla. I believe in the mission to ensure the Internet is a global public resource, open and accessible to all.",
        "github": "https://github.com/gabrielmicko",
        "id": "13",
        "linkedin": null,
        "name": "Gabriel Mičko",
        "talks": Array [
          Object {
            "date": "Aug 4 - 11:00",
            "duration": 150,
            "id": "13",
            "speakerId": "13",
            "talkTitle": "Building GraphQL Server with JS",
            "talkType": "workshop",
          },
        ],
        "twitter": "https://twitter.com/gabriel_micko",
      },
    ],
  },
}
`;
